src/path.c

```diff
@@ -369,6 +369,7 @@ void twin_path_arc_ellipse(twin_path_t *path,
                   param.extent);
 
     twin_path_set_matrix(path, save);
+    twin_path_draw(path, target_x, target_y);
```

Comment:
I didn't get it. Why?

Reply:
<img width="726" alt="image" src="https://github.com/user-attachments/assets/f31cbda0-722b-4e8c-9e59-44ff0363581f">

I think there might be some error in the arc ellipse calculation, so I enforced drawing to the target point to ensure correctness.

Reply:
> I think there might be some error in the arc ellipse calculation, so I enforced drawing to the target point to ensure correctness.

I would expect that we can overcome the underlying issues before the merging of this pull request.

Reply:
<img width="248" alt="image" src="https://github.com/user-attachments/assets/af35665e-bd21-492e-b89c-8d5bd9125b3f">

I think the accuracy of `twin_acos` could be improved. When I use the acos function from the standard math library, the results seem better. To enhance the accuracy of the arctan calculation, I try to use more precise angle values (0-32768) which allows us perform additional iterations for the CORDIC algorithm.

Reply:
@jouae, Please comment the above.

Reply:
@ndsl7109256 could you provide how you use math.h to get the conclusion, along with a comparison between math.h and current `twin_acos` ?

Reply:
I just replace `twin_acos` with `acos` in `vector_angle` and check the rendering result.Besides, I evaluated the CORDIC algorithm's error for different iteration count(original and modified) by testing all integer pairs $(x,y)$, $x \in [1,15]$, $y \in [1,15]$ For each pair, I calculated the arctangent using both the CORDIC method (with varying iterations) and the atan2 from math library.
| Iteration |    12    |    15    |
|:---------:|:--------:|:--------:|
| avg error | 0.071239 | 0.011101 |
|  std dev  | 0.054116 | 0.008201 |
| max error | 0.284898 | 0.043332 |

Reply:
@ndsl7109256 

I found you can compute the angle between two vectors without any `sqrt` operations.

Just modify `vector_angle` function as:
```diff
    twin_fixed_t dot = twin_fixed_mul(ux, vx) + twin_fixed_mul(uy, vy);
+   twin_fixed_t det = twin_fixed_mul(ux, vy) - twin_fixed_mul(uy, vx);
+   twin_fixed_t angle = twin_atan2(det, dot);
```

You can derive how to calculate the angle based on the following relationship

$$
\tan{\theta} = \frac{\sin{\theta}}{\cos{\theta}} = \frac{\text{det}(u,v)}{u \cdot v}
$$

$$
\cos{\theta} = \frac{u \cdot v}{|u||v|}
$$

$$
\sin{\theta} = \frac{\text{det}(u,v)}{|u||v|}
$$

where $\text{det}$ is the determinant of $u$ and $v$, for more detail see [Wikpedia: Determinant](https://en.wikipedia.org/wiki/Determinant#Geometric_meaning).

The advantage of this formula is that it avoids calculating the lengths of $u$ and $v$, thereby eliminating the need for the `sqrt` operation.

However, this might not be a solution to the rendering problem. The issue is likely caused by insufficient precision, leading to rendering errors. Further detailed analysis is required to obtain a more accurate conclusion and potential solutions.

Reply:
Your formula looks great, but the result looks bad :(
<img width="401" alt="image" src="https://github.com/user-attachments/assets/4baefe4f-7c4a-4afd-9867-44848940a1a9">
Maybe we could check it later